applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_fm_ale_solver.py

```diff
@@ -0,0 +1,190 @@
+
+        if self._IsPrintingRank():
+            KratosMultiphysics.Logger.PrintInfo("NavierStokesEmbeddedFMALEMonolithicSolver", "Construction of NavierStokesEmbeddedFMALEMonolithicSolver finished.")
+
+
+    def AddVariables(self):
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PRESSURE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.ACCELERATION)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_VELOCITY)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.IS_STRUCTURE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.BODY_FORCE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_H)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_AREA)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION_WATER_PRESSURE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NORMAL)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.EXTERNAL_PRESSURE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)              # Distance function nodal values
+        self.main_model_part.AddNodalSolutionStepVariable(KratosCFD.EMBEDDED_WET_PRESSURE)          # Post-process variable (stores the fluid nodes pressure and is set to 0 in the structure ones)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosCFD.EMBEDDED_WET_VELOCITY)          # Post-process variable (stores the fluid nodes velocity and is set to 0 in the structure ones)
+
+        if (self.fm_ale_step_frequency != 0):
+            self.virtual_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_DISPLACEMENT)
+            self.virtual_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_VELOCITY)
+
+        if self._IsPrintingRank():
+            KratosMultiphysics.Logger.PrintInfo("NavierStokesEmbeddedFMALEMonolithicSolver", "Fluid solver variables added correctly.")
+
+
+    def Initialize(self):
+        super(NavierStokesEmbeddedFMALEMonolithicSolver, self).Initialize()
+
+        ## Set the virtual model part geometry
+        if (self.fm_ale_step_frequency != 0):
+            self.mesh_moving_util.FillVirtualModelPart(self.main_model_part)     
+
+
+    def InitializeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            # Compute the BDF coefficients
+            (self.bdf_process).Execute()
+
+            #TODO: REMOVE THIS IF STATEMENT ONCE #1904 IS MERGED
+            step = self.main_model_part.ProcessInfo[KratosMultiphysics.STEP]
+            if(step + 1 >= self.GetMinimumBufferSize()):
```

Comment:
I believe you do not need this check here? It should be equivalent to the `if self._TimeBufferIsInitialized()` above.

Reply:
I forgot to remove this one... Thanks a lot!